### PR TITLE
Fix/issue 21 - Removing an UpGrade user.

### PR DESCRIPTION
**Pull request by YashilDepani**
In this PR, I have added the delete option for removing an UpGrade user. On clicking the delete-icon it prompts the dialogue box asking for confirmation. Type "delete" to confirm the delete.

image
image

### CODE REVIEW

File: frontend/projects/abtesting/src/app/core/users/store/users.effects.ts [hunk 1]
```diff
@@ -106,6 +106,18 @@ export class UsersEffects {
     )
   );
 
+  deleteUser$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(UsersActions.actionDeleteUser),
+      map(action => action.email),
+      filter(( email ) => !!email ),
```
Comment (JD2455) [inline frontend/projects/abtesting/src/app/core/users/store/users.effects.ts:113]: @YashilDepani  Can you please fix formatting here?

File: frontend/projects/abtesting/src/app/shared/components/delete/delete.component.scss [hunk 2]
```diff
@@ -0,0 +1,20 @@
+.dialog {
+    .name {
```
Comment (JD2455) [inline frontend/projects/abtesting/src/app/shared/components/delete/delete.component.scss:2]: @YashilDepani Can you please fix the indentation here?

File: frontend/projects/abtesting/src/app/shared/components/delete/delete.component.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+@Component({
+  selector: 'app-delete',
+  templateUrl: './delete.component.html',
+  styleUrls: ['./delete.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class DeleteComponent {
+
+  deleteConfirm: string;
+  message: boolean;
```
Comment (JD2455) [inline frontend/projects/abtesting/src/app/shared/components/delete/delete.component.ts:13]: @YashilDepani Can we update this variable name to more descriptive? may be  isDeleteButtonClicked?

File: frontend/projects/abtesting/src/app/features/dashboard/profile/components/profile-info/profile-info.component.ts [hunk 4]
```diff
@@ -161,6 +162,20 @@ export class ProfileInfoComponent implements OnInit, OnDestroy, AfterViewInit {
     });
   }
 
+  openDeleteUserModal(user: User) {
+    const dialogRef = this._matDialog.open(DeleteComponent, {
+      panelClass: 'delete-modal'
+    });
+
+    dialogRef.afterClosed().subscribe(result => {
```
Comment (JD2455) [inline frontend/projects/abtesting/src/app/features/dashboard/profile/components/profile-info/profile-info.component.ts:170]: @YashilDepani Can we update name of result variable? may be isDeleteButtonClicked?

File: frontend/projects/abtesting/src/app/features/dashboard/home/pages/view-experiment/view-experiment.component.ts [hunk 5]
```diff
@@ -101,13 +101,15 @@ export class ViewExperimentComponent implements OnInit, OnDestroy {
...
 
   deleteExperiment() {
-    const dialogRef = this.dialog.open(DeleteExperimentComponent, {
-      panelClass: 'delete-modal',
-      data: { experimentName: this.experiment.name, experimentId: this.experiment.id }
+    const dialogRef = this.dialog.open(DeleteComponent, {
+      panelClass: 'delete-modal'
     });
 
     dialogRef.afterClosed().subscribe(result => {
```
Comment (JD2455) [inline frontend/projects/abtesting/src/app/features/dashboard/home/pages/view-experiment/view-experiment.component.ts:108]: @YashilDepani Can we update name of result variable? may be isDeleteButtonClicked?

File: frontend/projects/abtesting/src/app/core/users/store/users.effects.ts (hunk 1 above)
Comment (YashilDepani) [reply frontend/projects/abtesting/src/app/core/users/store/users.effects.ts]: code updated

File: frontend/projects/abtesting/src/app/shared/components/delete/delete.component.scss (hunk 2 above)
Comment (YashilDepani) [reply frontend/projects/abtesting/src/app/shared/components/delete/delete.component.scss]: code updated

File: frontend/projects/abtesting/src/app/shared/components/delete/delete.component.ts (hunk 3 above)
Comment (YashilDepani) [reply frontend/projects/abtesting/src/app/shared/components/delete/delete.component.ts:13]: code updated

File: frontend/projects/abtesting/src/app/features/dashboard/profile/components/profile-info/profile-info.component.ts (hunk 4 above)
Comment (YashilDepani) [reply frontend/projects/abtesting/src/app/features/dashboard/profile/components/profile-info/profile-info.component.ts]: code updated

File: frontend/projects/abtesting/src/app/features/dashboard/home/pages/view-experiment/view-experiment.component.ts (hunk 5 above)
Comment (YashilDepani) [reply frontend/projects/abtesting/src/app/features/dashboard/home/pages/view-experiment/view-experiment.component.ts]: code updated

**Review by JD2455 — APPROVED** [review]

**Pull request merged**